extensions/cloud/portability-cloud-microsoft/src/main/java/org/dataportabilityproject/cloud/microsoft/cosmos/AzureKeyVaultStore.java

```diff
@@ -19,37 +16,42 @@
   private final String vaultUrl;
   private final KeyVaultClient vaultClient;
 
-  public AzureKeyVaultStore(String keyVaultName) {
+  public AzureKeyVaultStore(
+      String keyVaultName, String tenantId, String clientId, String clientSecret) {
     vaultUrl = String.format(VAULT_ADDRESS, keyVaultName);
 
-    // NB: env vars should be sourced in a secure form
-    // TODO should env vars be sourced from config?
-    String azureClientId = getenv("AZURE_CLIENT_ID");
-    Objects.requireNonNull(azureClientId, "AZURE_CLIENT_ID not set in environment");
-
-    String azureTenantId = getenv("AZURE_TENANT_ID");
-    Objects.requireNonNull(azureTenantId, "AZURE_TENANT_ID not set in environment");
-
-    String azureClientSecret = getenv("AZURE_CLIENT_SECRET");
-    Objects.requireNonNull(azureClientSecret, "AZURE_CLIENT_SECRET not set in environment");
-
     ApplicationTokenCredentials credentials =
-        new ApplicationTokenCredentials(
-            azureClientId, azureTenantId, azureClientSecret, AzureEnvironment.AZURE);
+        new ApplicationTokenCredentials(clientId, tenantId, clientSecret, AzureEnvironment.AZURE);
     vaultClient = new KeyVaultClient(credentials);
   }
 
   @Override
   public AppCredentials getAppCredentials(String keyName, String secretName) throws IOException {
-    SecretBundle secretBundle = vaultClient.getSecret(vaultUrl, "/secrets/" + keyName);
+    String normalizedKey = normalize(keyName);
+    SecretBundle keyBundle = vaultClient.getSecret(vaultUrl, normalizedKey);
+    if (keyBundle == null) {
+      throw new IOException(("Key not found: " + secretName));
+    }
+    String keyValue = keyBundle.value();
+
+    String normalizedSecret = normalize(secretName);
+    SecretBundle secretBundle = vaultClient.getSecret(vaultUrl, normalizedSecret);
     if (secretBundle == null) {
-      throw new IOException(("Secret not found: " + secretName));
+      throw new IOException(("Key not found: " + secretName));
     }
-    return new AppCredentials(keyName, secretBundle.value());
+    String secretValue = secretBundle.value();
+
+    return new AppCredentials(keyValue, secretValue);
   }
 
   public String getSecret(String key) {
-    SecretBundle secretBundle = vaultClient.getSecret(vaultUrl, "/secrets/" + key);
+    String normalizedKey = normalize(key);
+    SecretBundle secretBundle = vaultClient.getSecret(vaultUrl, normalizedKey);
     return secretBundle == null ? null : secretBundle.value();
   }
+
+  /** Normalizes a key to adhere to Azure Vault naming requirements. */
+  private String normalize(String key) {
```

Comment:
static

Reply:
Quick Q: why would this be static?

Reply:
Primarily since it doesn't depend on member variables, we can mark it as static and indicate that.